### PR TITLE
Only allow numeric text in number parameters.

### PR DESCRIPTION
**Pull request by OrionsGuardian**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Numeric fields will now check the input text against a number pattern to confirm valid numeric data.

### Alternate Designs

Could wait indefinitely for the component creator to provide a validation method.

### Why Should This Be In Core?

Improved User experience.

### Benefits

Prevents bad data being used in number fields.

### Possible Drawbacks

N/A

### Verification Process

Test on Integer parameters and Float parameters.
Confirm the non-numeric keystrokes are being ignored.

### Applicable Issues

#1687

### CODE REVIEW

**Review by antares1470 — APPROVED** [review]

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/NumberInputPane.java [hunk 1]
```diff
@@ -126,19 +126,30 @@ public NumberInputPane(final PluginParameter<?> parameter) {
         // Just typing doesn't fire value property change events, and doesn't allow us to change the style
         // when the string doesn't validate.
         field.getEditor().textProperty().addListener((final ObservableValue<? extends String> ov, final String oldValue, final String newValue) -> {
-            final String error = parameter.validateString(field.getValueFactory().getValue().toString());
-            if (error != null) {
-                tooltip.setText(error);
-                field.setTooltip(tooltip);
-                field.setId(INVALID_ID);
+            final int dotPos = newValue.indexOf(".");
```
Comment (antares1470) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/NumberInputPane.java:129]: `SeparatorConstants.PERIOD`

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/NumberInputPane.java [hunk 2]
```diff
@@ -126,19 +126,30 @@ public NumberInputPane(final PluginParameter<?> parameter) {
...
+            final String intPart = dotPos > -1 ? newValue.substring(0, dotPos) : newValue;
+            final String decPart = dotPos > -1 ? newValue.substring(dotPos + 1) : "";
+            final boolean isIntVal = parameter.getType().getId().equals(IntegerParameterType.ID);
+            // Integers: Max 9 digits.  Floats: Max 8 digits before the decimal, and 2 digits after.
+            if (intPart.matches("[\\-]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?" + (isIntVal ? "[0-9]?" : ""))
```
Comment (antares1470) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/NumberInputPane.java:134]: could the series of [0-9]?... be reduced to [0-9]{0, 8}?
I could also be wrong but with that regex statement, technically "-" on its own will match (so you might need to specify that if there is a "-" then there needs to be something after it)

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/NumberInputPane.java [hunk 3]
```diff
@@ -126,19 +126,30 @@ public NumberInputPane(final PluginParameter<?> parameter) {
...
-                tooltip.setText(error);
-                field.setTooltip(tooltip);
-                field.setId(INVALID_ID);
+            final int dotPos = newValue.indexOf(".");
+            final String intPart = dotPos > -1 ? newValue.substring(0, dotPos) : newValue;
+            final String decPart = dotPos > -1 ? newValue.substring(dotPos + 1) : "";
+            final boolean isIntVal = parameter.getType().getId().equals(IntegerParameterType.ID);
+            // Integers: Max 9 digits.  Floats: Max 8 digits before the decimal, and 2 digits after.
+            if (intPart.matches("[\\-]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?[0-9]?" + (isIntVal ? "[0-9]?" : "")) 
+                                && (decPart.isBlank() || (decPart.matches("[0-9]?[0-9]?") && !isIntVal))) {
```
Comment (antares1470) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/NumberInputPane.java:135]: similarly [0-9]{0,2}

**Review by antares1470 — CHANGES_REQUESTED** [review]
Apologies. I got excited after testing that I forgot to check the code 😄. See the below comments

**Review by Delphinus8821 — APPROVED** [review]

**Review by antares1470 — APPROVED** [review]

**Pull request merged**